tests/services/test_cases.py

```diff
@@ -213,3 +236,49 @@ def test_update_failure_raised_appropriate_custom_exception(
             cases_service.update(_TEST_CASE_NUMBER, findings=u"x")
 
         assert e.value.args[0] == u"Cannot update a closed case."
+
+    def test_create_when_fails_with_name_exists_error_raises_custom_exception(
+        self, mock_connection, mock_name_exists_response
+    ):
+        cases_service = CasesService(mock_connection)
+        mock_connection.post.side_effect = Py42BadRequestError(
+            mock_name_exists_response
+        )
+        with pytest.raises(Py42CaseNameExistsError) as e:
+            cases_service.create("Duplicate")
+
+        assert (
+            e.value.args[0]
+            == u"Case name 'Duplicate' already exists, please set another name"
+        )
+
+    def test_create_when_fails_with_description_too_long_error_raises_custom_exception(
+        self, mock_connection, mock_description_too_long_response
+    ):
+        cases_service = CasesService(mock_connection)
+        mock_connection.post.side_effect = Py42BadRequestError(
+            mock_description_too_long_response
+        )
+        with pytest.raises(Py42DescriptionLimitExceededError) as e:
+            cases_service.create("test", description=u"supposedly too long")
+
+        assert (
+            e.value.args[0]
+            == u"Description limit exceeded, max 254 characters allowed."
+        )
+
+    def test_update_when_fails_with_description_too_long_error_raises_custom_exception(
+        self, mock_connection, mock_get_response, mock_description_too_long_response
+    ):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.return_value = mock_get_response
+        mock_connection.put.side_effect = Py42BadRequestError(
+            mock_description_too_long_response
+        )
+        with pytest.raises(Py42DescriptionLimitExceededError) as e:
+            cases_service.update(_TEST_CASE_NUMBER, description=u"supposedly too long")
+
+        assert (
+            e.value.args[0]
+            == u"Description limit exceeded, max 254 characters allowed."
+        )
```

Comment:
Could add a test that makes sure that the 400 still gets raised when none of the scriptions match any expected custom-exception triggers.

Not super important, but code coverage wise, it is a missing covered section of the execution tree.

Reply:
Added.